### PR TITLE
feat: Timestamp support

### PR DESCRIPTION
**Pull request by ShubhamChaturvedi7**
*Issue #, if available:*

*Description of changes:*
feat: Timestamp support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java [hunk 1]
```diff
@@ -652,11 +652,17 @@ public String unionShape(final UnionShape shape) {
 
   @Override
   public String timestampShape(final TimestampShape shape) {
-    //TODO: This is a stub implementation and not working yet.
     writer.addImport("time");
```
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java:655]: Do you still need this import?

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java [hunk 2]
```diff
@@ -652,11 +652,17 @@ public String unionShape(final UnionShape shape) {
...
+    writer.addImportFromModule(DAFNY_RUNTIME_GO_LIBRARY_MODULE, "dafny");
+    var conversionCode =
+      "dafny.SeqOfChars([]dafny.Char(%s.Format(\"2006-01-02T15:04:05.999999Z\"))...)".formatted(
```
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java:658]: Did you manually test this wrapping/unwrapping with AWS SDK for Go?
AWS SDK <---> native conversion is not covered by the SimpleTimestamp TestModel.

I don't know how strict Go's `time` is; if AWS SDK for Go always reports time to milli precision, but the format expects micro precision, does it just fill in micro time as `000`?
Are we ok with the accuracy possibly not being correct? (i.e. Reporting microseconds when only millis were given. Might be a broader team question)

Comment (lucasmcdonald3) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java:658]: Chatted offline -- I would like to see a manual test that AWS SDK for Go returns timestamps that work with our code, but given
1. Deadlines
2. Our Dafny code doesn't interact with timestamps and we have ways to guard users from calling timestamps

this seems fine.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java (hunk 1 above)
Comment (lucasmcdonald3) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java:655]: (this is a nit)

**Review by lucasmcdonald3 — APPROVED** [review]

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java (hunk 1 above)
Comment (ShubhamChaturvedi7) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java:655]: Yup - need to use the `time.Time` package.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java (hunk 2 above)
Comment (ShubhamChaturvedi7) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java:658]: Yeah - our test deals with `Items` and items doesn't have time type. It uses string. To test this we'd need API like https://github.com/aws/aws-sdk-go-v2/blob/main/service/dynamodb/api_op_DescribeTable.go#L59 which is not in our scope - and as Lucas said, we have the safeguards as well.

**Pull request merged**